### PR TITLE
Block uploading blobs while getting pending state

### PR DESCRIPTION
**Pull request by dannimad**
## Description

[Bug](https://dev.azure.com/fluidframework/internal/_workitems/edit/5214)

Assert in sendBlobAttachOp is hitting. The stack trace indicates we are trying to resubmit a blobAttach op while not having its corresponding blob in the pending list. There could be a mismatch between the op pending list and the blob pending list now that the latter is async if we keep uploading blobs while being in shutdown phase. This PR blocks uploading blobs during that process.

### CODE REVIEW

File: packages/runtime/container-runtime/src/blobManager.ts [hunk 1]
```diff
@@ -1038,6 +1044,7 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 	}
 
 	public async getPendingBlobs(waitBlobsToAttach?: boolean): Promise<IPendingBlobs> {
+		this.blockUploading = true;
 		if (waitBlobsToAttach) {
 			await this.shutdownPendingBlobs();
```
Comment (anthony-murphy) [inline packages/runtime/container-runtime/src/blobManager.ts:1049]: can we reduce concepts. here we have block, wait, and shutdown, and they are all for basically the same thing, i'd rather them use the same terms.

**Comment by anthony-murphy** [comment]
nit; in the description you reference internal things, we should minimize this in our git hub, as may can't access them.  I prefer to reference at the bottom of the description, as it is less prominent, and use the `AB#<id>` format for only the issue. other items should be linked from the issue, and no the pr.

File: packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts [hunk 2]
```diff
@@ -1198,6 +1199,33 @@ describeNoCompat("stashed ops", (getTestObjectProvider) => {
 		);
 	});
 
+	it("uploading blob while closing", async function () {
```
[the remark this answers is not in the document]
Comment (anthony-murphy) [reply packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts:1202]: also, if i run this test it doesn't hit the assert, should it?

**Comment by dannimad** [comment]
I'm closing this PR given that this is a short term solution that we don't want long term. We should be able to keep uploading blobs while closing. I'll revisit this once other blob workflow fixes get done.

**Pull request closed without merging**